napari/layers/_shapes_layer/shape_list.py

```diff
@@ -0,0 +1,616 @@
+
+    Attributes
+    ----------
+    shapes : list
+        Length N list of N shape objects
+    _mesh_types : list
+        Length two list of the different mesh types corresponding to faces and
+        edges
+    _vertices : np.ndarray
+        Mx2 array of all vertices from all shapes
+    _index : np.ndarray
+        Length M array with the index (0, ..., N-1) of each shape that each
+        vertex corresponds to
+    _z_index : np.ndarray
+        Length N array with z_index of each shape
+    _z_order : np.ndarray
+        Length N array with z_order of each shape. This must be a permutation
+        of (0, ..., N-1).
+    _mesh_vertices : np.ndarray
+         Qx2 array of vertices of all triangles for shapes including edges and
+         faces
+    _mesh_vertices_centers : np.ndarray
+         Qx2 array of centers of vertices of triangles for shapes. For vertices
+         corresponding to faces these are the same as the actual vertices. For
+         vertices corresponding to edges these values should be added to a
+         scaled `_mesh_vertices_offsets` to get the actual vertex positions.
+         The scaling corresponds to the width of the edge
+    _mesh_vertices_offsets : np.ndarray
+         Qx2 array of offsets of vertices of triangles for shapes. For vertices
+         corresponding to faces these are 0. For vertices corresponding to
+         edges these values should be scaled and added to the
+         `_mesh_vertices_centers` to get the actual vertex positions.
+         The scaling corresponds to the width of the edge
+    _mesh_vertices_index : np.ndarray
+         Qx2 array of the index (0, ..., N-1) of each shape that each vertex
+         corresponds and the mesh type (0, 1) for face or edge.
+    _mesh_triangles : np.ndarray
+        Px3 array of vertex indices that form the mesh triangles
+    _mesh_triangles_index : np.ndarray
+        Px2 array of  the index (0, ..., N-1) of each shape that each triangle
+        corresponds and the mesh type (0, 1) for face or edge.
+    _mesh_triangles_colors : np.ndarray
+        Px4 array of the rgba color of each triangle
+    _mesh_triangles_z_order : np.ndarray
+        Length P array of the z order of each triangle. Must be a permutation
+        of (0, ..., P-1)
+    _shape_types : dict
+        Dictionary of supported shape types and their corresponding objects.
+    """
+    _mesh_types = ['face', 'edge']
+
+    _shape_types = ({'rectangle': Rectangle, 'ellipse': Ellipse, 'line': Line,
+                    'path': Path, 'polygon': Polygon})
+
+    def __init__(self, data, *, shape_type='rectangle', edge_width=1,
+                 edge_color='black', face_color='white', opacity=1, z_index=0):
+
+        self.shapes = []
+        self._vertices = np.empty((0, 2))
+        self._index = np.empty((0), dtype=int)
+        self._z_index = np.empty((0), dtype=int)
+        self._z_order = np.empty((0), dtype=int)
+
+        self._mesh_vertices = np.empty((0, 2))
```

Comment:
Here's another spot you could have a MeshProperties object

Reply:
I have added a `Mesh` object